Makefile

```diff
@@ -274,6 +274,10 @@ artifacts: kustomize yq helm ## Generate release artifacts.
 	# Revert the image changes
 	$(YQ)  e  '.controllerManager.manager.image.repository = "$(IMAGE_REGISTRY)/$(IMAGE_NAME)" | .controllerManager.manager.image.tag = "main" | .controllerManager.manager.image.pullPolicy = "Always"' -i charts/kueue/values.yaml
 	GO_BUILD_ENV="$(GO_BUILD_ENV)" GO_CMD="$(GO_CMD)" LD_FLAGS="$(LD_FLAGS)" BUILD_DIR="artifacts" BUILD_NAME=kubectl-kueue PLATFORMS="$(CLI_PLATFORMS)" ./hack/multiplatform-build.sh ./cmd/kueuectl/main.go
+	# Include kjobctl artifacts
+	cd cmd/experimental/kjobctl && make artifacts
+	mv cmd/experimental/kjobctl/artifacts/kubectl-kjob-* artifacts/
+	mv cmd/experimental/kjobctl/artifacts/manifests.yaml artifacts/kubectl-kjob-manifests.yaml
```

Comment:
I have some questions as I'm a bit concerned to add experimental artifacts and release along with the main Kueue artifacts. Especially as we are considering to move the kjobctl project to a dedicated kubernetes-sigs project.

1. is this PR part of the release process? If so, how is it envisioned as a whole?
2. how are the artifacts for the other CLI `kueuectl` built and released (I don't see the step here). could we follow the same pattern?
3. what is the build time of these steps?
4. should we consider a dedicated step like "artifacts-kjobctl" and invoke it during the release process

Reply:
@IrvingMg  Given the uncertainty of how kjobctl is going to be released, I would suggest to split the PR into two:
1. "[kjobctl] Build kjobctl executable artifacts" (without these lines, I'm happy to approve it now)
2. "[kjobctl] Expose the kjobctl executables as artifacts" (Just the lines, we can merge it closer to the 0.9 release date if confirmed this is part of Kueue)

Reply:
Ok, so should this PR be only to add the `make artifacts` directive to kjobctl's Makefile without including the binary in the Kueue artifacts?

Reply:
Yeah, let's start with this step

Reply:
Done. Removed changes in Kueue's Makefile and PR renamed.

Reply:
Cool I open the issue to decide on the release process: https://github.com/kubernetes-sigs/kueue/issues/2778